src/Umbraco.Core/Models/PublicAccessEntry.cs

```diff
@@ -107,9 +107,9 @@ public void RemoveRule(PublicAccessRule rule)
 
         public void ClearRules()
         {
-            foreach (var rule in _ruleCollection)
+            for (var i = _ruleCollection.Count - 1; i >= 0; i--)
```

Comment:
I replaced some code that made little sense by more code that makes even less sense. Fixing: all we need is `_ruleCollection.Clear();`